Services/Calendar/classes/class.ilCalendarSelectionBlockGUI.php

```diff
@@ -330,7 +330,12 @@ protected function getLegacyContent(): string
 
     protected function renderItem(array $a_set, ilTemplate $a_tpl): void
     {
-        if (strlen((string) $a_set['path'])) {
+        $obj_type = ilObject::_lookupType($a_set['obj_id']);
+
+        if (
+            strlen((string) $a_set['path']) &&
+            $this->obj_def->isAllowedInRepository($obj_type)
```

Comment:
Is it possible to move this statement to ilCAlendarSelectionBlockGUI::buildPath?